workbase/src/renderer/components/SchemaDesign/RightBar/SettingsTab/DisplaySettingsPanel.vue

```diff
@@ -0,0 +1,100 @@
+<template>
+    <div class="panel-container noselect">
+        <div @click="showConceptDisplayContent = !showConceptDisplayContent" class="panel-header">
+            <vue-icon :icon="(showConceptDisplayContent) ?  'chevron-down' : 'chevron-right'" iconSize="14" className="vue-icon"></vue-icon>
+            <h1>Display Settings</h1>
+        </div>
+        <div v-show="showConceptDisplayContent">
+            <div class="panel-content" v-if="!currentKeyspace">
+                Please select a keyspace
+            </div>
+
+            <div class="panel-content" v-else>
+                <div class="panel-content-item">
+                    <h1 class="panel-label">Entity types:</h1>
+                    <div class="panel-value load-roleplayers-switch"><vue-switch :isToggled="showEntities" v-on:toggled="showEntities = !showEntities"></vue-switch></div>
+                </div>
+                <div class="panel-content-item">
+                    <h1 class="panel-label">Attribute Types:</h1>
+                    <div class="panel-value load-roleplayers-switch"><vue-switch :isToggled="showAttributes" v-on:toggled="showAttributes = !showAttributes"></vue-switch></div>
+                </div>
+                <div class="panel-content-item">
+                    <h1 class="panel-label">Relationship Types:</h1>
+                    <div class="panel-value load-roleplayers-switch"><vue-switch :isToggled="showRelationships" v-on:toggled="showRelationships = !showRelationships"></vue-switch></div>
+                </div>
+        
+                
+            </div>
+        </div>
+    </div>
+</template>
+
+<script>
+  import { createNamespacedHelpers } from 'vuex';
+
+  export default {
+    name: 'DisplaySettingsPanel',
+    components: { },
+    data() {
+      return {
+        showConceptDisplayContent: true,
+        showEntities: true,
+        showAttributes: true,
+        showRelationships: true,
+      };
+    },
+    beforeCreate() {
+      const { mapGetters } = createNamespacedHelpers('schema-design');
+
+      // computed
+      this.$options.computed = {
+        ...(this.$options.computed || {}),
+        ...mapGetters(['currentKeyspace']),
+      };
+    },
+    watch: {
+      showEntities(val) {
+        this.showTypes(val, 'ENTITY_TYPE');
+      },
+      showAttributes(val) {
+        this.showTypes(val, 'ATTRIBUTE_TYPE');
+      },
+      showRelationships(val) {
+        this.showTypes(val, 'RELATIONSHIP_TYPE');
+      },
+    },
+    methods: {
+      showTypes(showType, baseType) {
+        const schemaConcepts = this.$store.state['schema-design'].visFacade.getAllNodes().filter((x => x.baseType === baseType)).map(x => Object.assign(x, { hidden: !showType }));
+        this.$store.state['schema-design'].visFacade.updateNode(schemaConcepts);
```

Comment:
to we always access the schema design state using: `this.$store.state['schema-design']`?

Reply:
ah no. We can use a getter. Change has been done.